### ISSUE TITLE
Prefix the "ytdl-format" argument value by "best[height<=${opts.maxheight}]/"

### ISSUE THREAD

**Issue by sebma** [Thann/play-with-mpv#35]
Hi,
Can you please update the `--ytdl-format` argument value in your [../chrome-extension/background.js](../tree/master/chrome-extension/background.js) according to this `git diff` output so that format selection on "https://ok.ru/video" work : 
```diff
$ git br 
* master
$ git describe --tags --abbrev=0
0.1.0
$ git diff chrome-extension/background.js
diff --git a/chrome-extension/background.js b/chrome-extension/background.js
index 799d99d..4de29e8 100644
--- a/chrome-extension/background.js
+++ b/chrome-extension/background.js
@@ -25,7 +25,7 @@ function playUrl(url, pause) {
                }
                if (opts.maxheight) {
                        opts.mpv_args.splice(0, 0,
-                       `--ytdl-format=bestvideo[height<=?${opts.maxheight}]+bestaudio/best`);
+                       `--ytdl-format=best[height<=${opts.maxheight}]/bestvideo[height<=?${opts.maxheight}]+bestaudio/best`);
                }
                const query = (`?play_url=` + encodeURIComponent(url) + [''].concat(
                        opts.mpv_args.map(encodeURIComponent)).join('&mpv_args='));

```

**Comment by Chaostheorie** [comment]
Kinda late but would you mind if I open a pull request with your changes?

**Comment by sebma** [comment]
@Chaostheorie Thanks. Also please don't forget to check if this commit works when playing videos on https://ok.ru/video with the current `youtube-dl` version.

**Comment by Haxprox** [comment]
In addition to this, would it be great to have a preferred hardware codec option like 'avc1, hevc, vp9/8, etc?
`bestvideo[vcodec^=avc1|hevc|vp9|vp8]`
Maybe somehow here: https://github.com/Thann/play-with-mpv/blob/master/chrome-extension/options.html with an additional and static field?
But the perfect solution of course is a dynamic option when pressing the button

**Comment by Chaostheorie** [comment]
This sounds more like something you would configure with mpv rather than handle with this plugin. Are you sure this is url dependent and can't be passed to mpv/ youtube-dl with a config?

**Comment by Haxprox** [comment]
Yup, totally agree with you but thinking about size as a static option, codec option can solve other hardware issues when the user experience with lagging by unknown reasons and can choose some additional codec options. But another hand, yes, it's just for the advanced users, of course.
For example:
Having an old intel processor, the hardware is able to handle only h.264 and if I pass '[bestvideo]' options without arguments, I'll get VP9 codec with lagging picture as you actually know :)

**Comment by Chaostheorie** [comment]
Huh, I didn't consider this. Your point is indeed something that should be considered with respect for different setups.
I guess it would be useful to have an option for the extension instead of having this toggled by default to not hurt UX. I will look into integrating this tomorrow/ on the weekend.
Thank you for bringing this to my attention.

**Comment by Haxprox** [comment]
Brilliant, thank you!
The example below is just for your additional attention and review:
1. Pointing to 4K video and default '[bestvideo]' option we're getting the next result:

> ~ ➤ mpv --start=1:40 --pause https://www.youtube.com/watch\?v\=LXb3EKWsInQ --ytdl-format="**bestvideo**+bestaudio/best"                                                                                               
> Resuming playback. This behavior can be disabled with --no-resume-playback.
>  (+) Video --vid=1 (*) **(av1 3840x2160 59.940fps)**
>  (+) Audio --aid=1 --alang=eng (*) (opus 2ch 48000Hz)
> AO: [pulse] 48000Hz stereo 2ch float
> VO: [gpu] 3840x2160 yuv420p10
> (Paused) AV: 00:01:40 / 00:05:13 (32%) A-V:  0.000 Cache: 5.9s/20MB
> Saving state.
> 
> Exiting... (Quit)

2. For H.264 option:

> ~ ➤ mpv --start=1:40 --pause https://www.youtube.com/watch\?v\=LXb3EKWsInQ --ytdl-format="**bestvideo[vcodec^=avc1]**+bestaudio/best"                                                                                 
> Resuming playback. This behavior can be disabled with --no-resume-playback.
>  (+) Video --vid=1 (*) **(h264 1920x1080 59.940fps)**
>  (+) Audio --aid=1 --alang=eng (*) (opus 2ch 48000Hz)
> [vaapi] libva: /usr/lib64/dri/iHD_drv_video.so init failed
> Cannot load libcuda.so.1
> AO: [pulse] 48000Hz stereo 2ch float
> Using hardware decoding (vaapi).
> VO: [gpu] 1920x1080 vaapi[nv12]
> (Paused) AV: 00:01:40 / 00:05:13 (32%) A-V:  0.000 Cache: 22s/14MB
> Saving state.
> 
> Exiting... (Quit) 

3. And for VP9:

> ~ ➤ mpv --start=1:40 --pause https://www.youtube.com/watch\?v\=LXb3EKWsInQ --ytdl-format="**bestvideo[vcodec^=vp9]**+bestaudio/best"                                                                                  
> Resuming playback. This behavior can be disabled with --no-resume-playback.
>  (+) Video --vid=1 (*) **(vp9 3840x2160 59.940fps)**
>  (+) Audio --aid=1 --alang=eng (*) (opus 2ch 48000Hz)
> [vaapi] libva: /usr/lib64/dri/iHD_drv_video.so init failed
> Cannot load libcuda.so.1
> AO: [pulse] 48000Hz stereo 2ch float
> [ffmpeg/video] vp9: No support for codec vp9 profile 2.
> VO: [gpu] 3840x2160 yuv420p10
> (Paused) AV: 00:01:40 / 00:05:13 (32%) A-V:  0.000 Cache: 2.2s/7MB
> Saving state.
> 
> Exiting... (Quit)

By default, it returns VP1 which can't be supported by the older hardwares.
Most probably a good solution for the default option is H.264, it will get rid of future issues requests :)

**Comment by beew** [comment]
Can you just add this line to your mpv config file?
```
ytdl-format=bestvideo[height<=?1080][fps<=?30][vcodec!=?vp9]+bestaudio/best
```

Works for me.
